client.go

```diff
@@ -0,0 +1,593 @@
+package milter
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/emersion/go-message/textproto"
+)
+
+// Client is a wrapper for managing milter connections.
+//
+// Currently it just creates new connections using provided Dialer.
+type Client struct {
+	// Dialer is used to establish new connections to the milter.
+	// Set to empty net.Dialer{} by NewClient.
+	Dialer interface {
+		Dial(network string, addr string) (net.Conn, error)
+	}
+
+	ReadTimeout  time.Duration
+	WriteTimeout time.Duration
+
+	Network string
+	Address string
+}
+
+// NewClient creates a new Client instance populating fields with default
+// values.
+//
+// Dialer is net.Dialer with 10 second timeout, Read and Write timeouts are set
+// to 10 seconds too.
+func NewClient(network, address string) *Client {
```

Comment:
I'd expect a `NewClient` function to not set any timeout.

Reply:
I expect NewClient to provide Client instance with reasonable defaults set. milter implementations are generally located on the same system or on the same LAN, if it takes a lot of time to connect - it is likely a configuration problem.

Reply:
Then can we rename it to `NewDefaultClient` (following wording from the stdlib, see e.g.  https://golang.org/pkg/net/http/#DefaultTransport)?

Reply:
But if we switch to taking a Dialer, this won't matter anyway, see below